kaldb/src/test/java/com/slack/kaldb/metadata/zookeeper/ZooKeeperCachedMetadataStoreImplTest.java

```diff
@@ -368,62 +382,76 @@ public void stateChanged(CuratorFramework client, ConnectionState newState) {
     SnapshotMetadata snapshot2 = makeSnapshot("test2");
     assertThat(metadataStore.create(node2, serDe.toJsonStr(snapshot2), true).get()).isNull();
     await().untilAsserted(() -> assertThat(cache.get("node2").get()).isEqualTo(snapshot2));
-    assertThat(cache.getInstances()).containsOnly(snapshot1, snapshot2);
+    assertThat(cache.getInstances()).containsOnly(snapshot1, snapshot2, defaultRoot);
     assertThat(metadataStore.exists(node2).get()).isTrue();
     assertThat(metadataStore.get(node2).get()).isEqualTo(serDe.toJsonStr(snapshot2));
     assertThat(getCount(CACHE_ERROR_COUNTER, meterRegistry)).isEqualTo(1);
+    expectedCacheErrorCount = 1;
   }
 
   @SuppressWarnings("OptionalGetWithoutIsPresent")
   @Test
   public void testWatchingPersistentNode() throws Exception {
     final String root = "/root";
-    assertThat(metadataStore.create(root, "", true).get()).isNull();
+    assertThat(metadataStore.create(root, defaultRootStr, true).get()).isNull();
 
     final String node = "/root/node";
     SnapshotMetadata snapshot1 = makeSnapshot("test1");
     assertThat(metadataStore.create(node, serDe.toJsonStr(snapshot1), true).get()).isNull();
 
-    CachedMetadataStore<SnapshotMetadata> cache = makeCachedStore(root, null, serDe);
+    // Null listener works.
+    ZooKeeperCachedMetadataStore<SnapshotMetadata> cache = makeCachedStore(root, null, serDe);
 
-    await().untilAsserted(() -> assertThat(cache.get("node").get()).isEqualTo(snapshot1));
-    assertThat(cache.getInstances().size()).isEqualTo(1);
+    await().untilAsserted(() -> assertThat(cache.getInstances().size()).isEqualTo(2));
+    assertThat(cache.get("node").get()).isEqualTo(snapshot1);
+    assertThat(cache.get(root).get()).isEqualTo(defaultRoot);
+    assertThat(cache.getInstances().size()).isEqualTo(2);
     assertThat(metadataStore.exists(node).get()).isTrue();
     assertThat(metadataStore.get(node).get()).isEqualTo(serDe.toJsonStr(snapshot1));
 
     SnapshotMetadata snapshot11 = makeSnapshot("test11");
     assertThat(metadataStore.put(node, serDe.toJsonStr(snapshot11)).get()).isNull();
     await().untilAsserted(() -> assertThat(cache.get("node").get()).isEqualTo(snapshot11));
-    assertThat(cache.getInstances().size()).isEqualTo(1);
+    assertThat(cache.getInstances().size()).isEqualTo(2);
+    assertThat(cache.getInstances()).containsOnly(snapshot11, defaultRoot);
 
     assertThat(metadataStore.delete(node).get()).isNull();
     await().untilAsserted(() -> assertThat(cache.get(node).isPresent()).isFalse());
-    assertThat(cache.getInstances()).isEmpty();
+    assertThat(cache.getInstances()).containsOnly(defaultRoot);
 
     cache.close();
+    await()
+        .untilAsserted(
+            () ->
+                assertThat(((ZooKeeperCachedMetadataStoreImpl<SnapshotMetadata>) cache).isStopped())
+                    .isTrue());
+    // TODO: Get instances after cache close works?
```

Comment:
This todo can be removed, as this behavior is was intentional - see `ZooKeeperCachedMetadataStoreImplTest.watchCachePersistentTreeTest()`